miner/worker.go

```diff
@@ -245,9 +246,19 @@ func (miner *Miner) prepareWork(genParams *generateParams, witness bool) (*envir
 	}
 	if genParams.gasLimit != nil { // override gas limit if specified
 		header.GasLimit = *genParams.gasLimit
-	} else if miner.chain.Config().Optimism != nil && miner.config.GasCeil != 0 {
-		// configure the gas limit of pending blocks with the miner gas limit config when using optimism
-		header.GasLimit = miner.config.GasCeil
+	} else if miner.chain.Config().Optimism != nil {
+		if miner.config.GasCeil != 0 {
+			// configure the gas limit of pending blocks with the miner gas limit config when using optimism
+			header.GasLimit = miner.config.GasCeil
+		}
+		if miner.chainConfig.IsHolocene(header.Time) {
+			if genParams.eip1559Params == nil {
+				return nil, errors.New("Expected eip1559 params, got none")
+			}
+			header.Nonce = *genParams.eip1559Params
+		} else if genParams.eip1559Params != nil {
+			return nil, errors.New("Got eip1559 params, expected none")
+		}
```

Comment:
This works because the `if genParams.eip1559Params == nil` check is nested.

Do we have this bit tested?

Reply:
no, unfortunately there were no existing optimism specific tests to build on here so I punted :) but let me look into adding.

Reply:
done